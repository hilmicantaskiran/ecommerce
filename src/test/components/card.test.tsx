import React from 'react';
import { render } from '@testing-library/react';
import Card from '../../components/card';
import axios from 'axios';
import { MemoryRouter } from 'react-router-dom';

jest.mock('axios');
const dummyProduct = {
    id: 1,
    name: 'Product 1',
    price: 100,
    image: 'https://via.placeholder.com/150',
    brand: 'Brand 1',
    model: 'Model 1',
    description: 'Description 1',
    createdAt: '2022-01-01'
}

describe('Card', () => {
    it('should render correctly', () => {
        (axios.get as jest.Mock).mockResolvedValue({ data: [dummyProduct] });
        const { asFragment } = render(
            <MemoryRouter>
                <Card 
                    key={dummyProduct.id} 
                    id={dummyProduct.id}
                    image={dummyProduct.image}
                    price={dummyProduct.price}
                    brand={dummyProduct.brand}
                    model={dummyProduct.model}
                    addToCart={() => {}}
                />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    }),
    it('should render correctly with no image', () => {
        (axios.get as jest.Mock).mockResolvedValue({ data: [{ ...dummyProduct, image: '' }] });
        const { asFragment } = render(
            <MemoryRouter>
                <Card 
                    key={dummyProduct.id} 
                    id={dummyProduct.id}
                    image={dummyProduct.image}
                    price={dummyProduct.price}
                    brand={dummyProduct.brand}
                    model={dummyProduct.model}
                    addToCart={() => {}}
                />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    }),
    it('should render correctly with no description', () => {
        (axios.get as jest.Mock).mockResolvedValue({ data: [{ ...dummyProduct, description: '' }] });
        const { asFragment } = render(
            <MemoryRouter>
                <Card 
                    key={dummyProduct.id} 
                    id={dummyProduct.id}
                    image={dummyProduct.image}
                    price={dummyProduct.price}
                    brand={dummyProduct.brand}
                    model={dummyProduct.model}
                    addToCart={() => {}}
                />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    }),
    it('should render correctly with no brand', () => {
        (axios.get as jest.Mock).mockResolvedValue({ data: [{ ...dummyProduct, brand: '' }] });
        const { asFragment } = render(
            <MemoryRouter>
                <Card 
                    key={dummyProduct.id} 
                    id={dummyProduct.id}
                    image={dummyProduct.image}
                    price={dummyProduct.price}
                    brand={dummyProduct.brand}
                    model={dummyProduct.model}
                    addToCart={() => {}}
                />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    }),
    it('should render correctly with no model', () => {
        (axios.get as jest.Mock).mockResolvedValue({ data: [{ ...dummyProduct, model: '' }] });
        const { asFragment } = render(
            <MemoryRouter>
                <Card 
                    key={dummyProduct.id} 
                    id={dummyProduct.id}
                    image={dummyProduct.image}
                    price={dummyProduct.price}
                    brand={dummyProduct.brand}
                    model={dummyProduct.model}
                    addToCart={() => {}}
                />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    }),
    it('should render correctly with no price', () => {
        (axios.get as jest.Mock).mockResolvedValue({ data: [{ ...dummyProduct, price: 0 }] });
        const { asFragment } = render(
            <MemoryRouter>
                <Card 
                    key={dummyProduct.id} 
                    id={dummyProduct.id}
                    image={dummyProduct.image}
                    price={dummyProduct.price}
                    brand={dummyProduct.brand}
                    model={dummyProduct.model}
                    addToCart={() => {}}
                />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
    it('should render correctly with no brand and model', () => {
        (axios.get as jest.Mock).mockResolvedValue({ data: [{ ...dummyProduct, brand: '', model: '' }] });
        const { asFragment } = render(
            <MemoryRouter>
                <Card 
                    key={dummyProduct.id} 
                    id={dummyProduct.id}
                    image={dummyProduct.image}
                    price={dummyProduct.price}
                    brand={dummyProduct.brand}
                    model={dummyProduct.model}
                    addToCart={() => {}}
                />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    }),
    it('should render correctly with no brand and model and price', () => {
        (axios.get as jest.Mock).mockResolvedValue({ data: [{ ...dummyProduct, brand: '', model: '', price: 0 }] });
        const { asFragment } = render(
            <MemoryRouter>
                <Card 
                    key={dummyProduct.id} 
                    id={dummyProduct.id}
                    image={dummyProduct.image}
                    price={dummyProduct.price}
                    brand={dummyProduct.brand}
                    model={dummyProduct.model}
                    addToCart={() => {}}
                />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    }),
    it('should render correctly with no brand and model and price and image', () => {
        (axios.get as jest.Mock).mockResolvedValue({ data: [{ ...dummyProduct, brand: '', model: '', price: 0, image: '' }] });
        const { asFragment } = render(
            <MemoryRouter>
                <Card 
                    key={dummyProduct.id} 
                    id={dummyProduct.id}
                    image={dummyProduct.image}
                    price={dummyProduct.price}
                    brand={dummyProduct.brand}
                    model={dummyProduct.model}
                    addToCart={() => {}}
                />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

