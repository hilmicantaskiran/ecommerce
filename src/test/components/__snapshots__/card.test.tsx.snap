// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center sm:h-72 bg-white p-2 drop-shadow"
  >
    <a
      class="flex flex-col group"
      href="/product/1"
      id="product"
    >
      <div
        class="w-full aspect-w-1 aspect-h-1 bg-gray-200 overflow-hidden xl:aspect-w-8 xl:aspect-h-10"
      >
        <img
          alt="product"
          class="w-full h-full object-center object-cover group-hover:opacity-75"
          src="https://via.placeholder.com/150"
        />
      </div>
      <h3
        class="mt-4 font-medium text-sm text-[#2a59fe]"
      >
        100₺
      </h3>
      <p
        class="mt-1 font-medium text-gray-900"
      >
        Brand 1 Model 1
      </p>
    </a>
    <button
      class="w-full mt-10 sm:mt-auto bg-[#2a59fe] text-white py-2 px-4 rounded-lg"
      id="addToCart"
    >
      Add to Cart
    </button>
  </div>
</DocumentFragment>
`;

exports[`Card should render correctly with no brand 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center sm:h-72 bg-white p-2 drop-shadow"
  >
    <a
      class="flex flex-col group"
      href="/product/1"
      id="product"
    >
      <div
        class="w-full aspect-w-1 aspect-h-1 bg-gray-200 overflow-hidden xl:aspect-w-8 xl:aspect-h-10"
      >
        <img
          alt="product"
          class="w-full h-full object-center object-cover group-hover:opacity-75"
          src="https://via.placeholder.com/150"
        />
      </div>
      <h3
        class="mt-4 font-medium text-sm text-[#2a59fe]"
      >
        100₺
      </h3>
      <p
        class="mt-1 font-medium text-gray-900"
      >
        Brand 1 Model 1
      </p>
    </a>
    <button
      class="w-full mt-10 sm:mt-auto bg-[#2a59fe] text-white py-2 px-4 rounded-lg"
      id="addToCart"
    >
      Add to Cart
    </button>
  </div>
</DocumentFragment>
`;

exports[`Card should render correctly with no brand and model 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center sm:h-72 bg-white p-2 drop-shadow"
  >
    <a
      class="flex flex-col group"
      href="/product/1"
      id="product"
    >
      <div
        class="w-full aspect-w-1 aspect-h-1 bg-gray-200 overflow-hidden xl:aspect-w-8 xl:aspect-h-10"
      >
        <img
          alt="product"
          class="w-full h-full object-center object-cover group-hover:opacity-75"
          src="https://via.placeholder.com/150"
        />
      </div>
      <h3
        class="mt-4 font-medium text-sm text-[#2a59fe]"
      >
        100₺
      </h3>
      <p
        class="mt-1 font-medium text-gray-900"
      >
        Brand 1 Model 1
      </p>
    </a>
    <button
      class="w-full mt-10 sm:mt-auto bg-[#2a59fe] text-white py-2 px-4 rounded-lg"
      id="addToCart"
    >
      Add to Cart
    </button>
  </div>
</DocumentFragment>
`;

exports[`Card should render correctly with no brand and model and price 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center sm:h-72 bg-white p-2 drop-shadow"
  >
    <a
      class="flex flex-col group"
      href="/product/1"
      id="product"
    >
      <div
        class="w-full aspect-w-1 aspect-h-1 bg-gray-200 overflow-hidden xl:aspect-w-8 xl:aspect-h-10"
      >
        <img
          alt="product"
          class="w-full h-full object-center object-cover group-hover:opacity-75"
          src="https://via.placeholder.com/150"
        />
      </div>
      <h3
        class="mt-4 font-medium text-sm text-[#2a59fe]"
      >
        100₺
      </h3>
      <p
        class="mt-1 font-medium text-gray-900"
      >
        Brand 1 Model 1
      </p>
    </a>
    <button
      class="w-full mt-10 sm:mt-auto bg-[#2a59fe] text-white py-2 px-4 rounded-lg"
      id="addToCart"
    >
      Add to Cart
    </button>
  </div>
</DocumentFragment>
`;

exports[`Card should render correctly with no brand and model and price and image 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center sm:h-72 bg-white p-2 drop-shadow"
  >
    <a
      class="flex flex-col group"
      href="/product/1"
      id="product"
    >
      <div
        class="w-full aspect-w-1 aspect-h-1 bg-gray-200 overflow-hidden xl:aspect-w-8 xl:aspect-h-10"
      >
        <img
          alt="product"
          class="w-full h-full object-center object-cover group-hover:opacity-75"
          src="https://via.placeholder.com/150"
        />
      </div>
      <h3
        class="mt-4 font-medium text-sm text-[#2a59fe]"
      >
        100₺
      </h3>
      <p
        class="mt-1 font-medium text-gray-900"
      >
        Brand 1 Model 1
      </p>
    </a>
    <button
      class="w-full mt-10 sm:mt-auto bg-[#2a59fe] text-white py-2 px-4 rounded-lg"
      id="addToCart"
    >
      Add to Cart
    </button>
  </div>
</DocumentFragment>
`;

exports[`Card should render correctly with no description 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center sm:h-72 bg-white p-2 drop-shadow"
  >
    <a
      class="flex flex-col group"
      href="/product/1"
      id="product"
    >
      <div
        class="w-full aspect-w-1 aspect-h-1 bg-gray-200 overflow-hidden xl:aspect-w-8 xl:aspect-h-10"
      >
        <img
          alt="product"
          class="w-full h-full object-center object-cover group-hover:opacity-75"
          src="https://via.placeholder.com/150"
        />
      </div>
      <h3
        class="mt-4 font-medium text-sm text-[#2a59fe]"
      >
        100₺
      </h3>
      <p
        class="mt-1 font-medium text-gray-900"
      >
        Brand 1 Model 1
      </p>
    </a>
    <button
      class="w-full mt-10 sm:mt-auto bg-[#2a59fe] text-white py-2 px-4 rounded-lg"
      id="addToCart"
    >
      Add to Cart
    </button>
  </div>
</DocumentFragment>
`;

exports[`Card should render correctly with no image 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center sm:h-72 bg-white p-2 drop-shadow"
  >
    <a
      class="flex flex-col group"
      href="/product/1"
      id="product"
    >
      <div
        class="w-full aspect-w-1 aspect-h-1 bg-gray-200 overflow-hidden xl:aspect-w-8 xl:aspect-h-10"
      >
        <img
          alt="product"
          class="w-full h-full object-center object-cover group-hover:opacity-75"
          src="https://via.placeholder.com/150"
        />
      </div>
      <h3
        class="mt-4 font-medium text-sm text-[#2a59fe]"
      >
        100₺
      </h3>
      <p
        class="mt-1 font-medium text-gray-900"
      >
        Brand 1 Model 1
      </p>
    </a>
    <button
      class="w-full mt-10 sm:mt-auto bg-[#2a59fe] text-white py-2 px-4 rounded-lg"
      id="addToCart"
    >
      Add to Cart
    </button>
  </div>
</DocumentFragment>
`;

exports[`Card should render correctly with no model 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center sm:h-72 bg-white p-2 drop-shadow"
  >
    <a
      class="flex flex-col group"
      href="/product/1"
      id="product"
    >
      <div
        class="w-full aspect-w-1 aspect-h-1 bg-gray-200 overflow-hidden xl:aspect-w-8 xl:aspect-h-10"
      >
        <img
          alt="product"
          class="w-full h-full object-center object-cover group-hover:opacity-75"
          src="https://via.placeholder.com/150"
        />
      </div>
      <h3
        class="mt-4 font-medium text-sm text-[#2a59fe]"
      >
        100₺
      </h3>
      <p
        class="mt-1 font-medium text-gray-900"
      >
        Brand 1 Model 1
      </p>
    </a>
    <button
      class="w-full mt-10 sm:mt-auto bg-[#2a59fe] text-white py-2 px-4 rounded-lg"
      id="addToCart"
    >
      Add to Cart
    </button>
  </div>
</DocumentFragment>
`;

exports[`Card should render correctly with no price 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center sm:h-72 bg-white p-2 drop-shadow"
  >
    <a
      class="flex flex-col group"
      href="/product/1"
      id="product"
    >
      <div
        class="w-full aspect-w-1 aspect-h-1 bg-gray-200 overflow-hidden xl:aspect-w-8 xl:aspect-h-10"
      >
        <img
          alt="product"
          class="w-full h-full object-center object-cover group-hover:opacity-75"
          src="https://via.placeholder.com/150"
        />
      </div>
      <h3
        class="mt-4 font-medium text-sm text-[#2a59fe]"
      >
        100₺
      </h3>
      <p
        class="mt-1 font-medium text-gray-900"
      >
        Brand 1 Model 1
      </p>
    </a>
    <button
      class="w-full mt-10 sm:mt-auto bg-[#2a59fe] text-white py-2 px-4 rounded-lg"
      id="addToCart"
    >
      Add to Cart
    </button>
  </div>
</DocumentFragment>
`;
